src/rewrite.rs

```diff
@@ -93,27 +99,95 @@ fn extract_scalar_function(expr: &Expr) -> Option<&ScalarFunction> {
     }
 }
 
+#[derive(Debug, Clone, Copy)]
+enum JsonOperator {
+    Arrow,
+    LongArrow,
+    Question,
+}
+
+impl TryFrom<&BinaryOperator> for JsonOperator {
+    type Error = DataFusionError;
+
+    fn try_from(op: &BinaryOperator) -> Result<Self> {
+        match op {
+            BinaryOperator::Arrow => Ok(JsonOperator::Arrow),
+            BinaryOperator::LongArrow => Ok(JsonOperator::LongArrow),
+            BinaryOperator::Question => Ok(JsonOperator::Question),
+            _ => Err(DataFusionError::Internal(format!(
+                "Unexpected operator {:?} in JSON function rewriter",
+                op
+            ))),
+        }
+    }
+}
```

Comment:
Looks like the `Error` is never really used, so let's just make it trivial.

Reply:
yup was about to push that 🧠